src/services/publishing/utils/utils.tsx

```diff
@@ -0,0 +1,21 @@
+import { Contract } from "ethers";
+import { getLogger } from "../../../utils/logger";
+
+const { error } = getLogger("services:supportsinterface");
+
+interface Erc165Contract extends Contract {
+  supportsInterface: (interfaceId: string) => Promise<boolean>;
+}
+
+export const supportsInterface = async (
+  contractInstance: Erc165Contract,
+  interfaceId: string
+): Promise<boolean> => {
+  let isSameInterfaceType = false;
+  try {
+    isSameInterfaceType = await contractInstance.supportsInterface(interfaceId);
+  } catch (supportsInterfaceErrorMessage) {
+    error(supportsInterfaceErrorMessage);
+  }
```

Comment:
Why are we catching this? Does it throw when there the `supportInterface` method is not found? Does it throw when the interface is not supported? 

If its the case where it throws when the method is not found, u might want to catch that and return false, otherwise rethrow the error.

Reply:
This throws an error when `suppoprtInterface` is not found, when the address is given is not a contract. if not supported would return false. 
 Will change this to return false if `call reverts exception` (method not found) else rethrow

Reply:
Yep good idea